test/spark/app/src/main/scala/Sonnets.scala

```diff
@@ -7,15 +7,19 @@ object Sonnets {
       .appName("SonnetsApp")
       .getOrCreate()
     val input = args.applyOrElse(0, (_: Any) => "s3a://example/main/sonnets.txt")
-    val output = args.applyOrElse(1, (_: Any) => "s3a://example/main/sonnets-wordcount.csv")
+    val output = args.applyOrElse(1, (_: Any) => "s3a://example/main/sonnets-wordcount")
     val sc = spark.sparkContext
     sc.setLogLevel("INFO")
     import spark.implicits._
     val sonnets = sc.textFile(input)
     val counts = sonnets.flatMap(line => line.split(" ")).map(word => (word, 1)).reduceByKey(_ + _).toDF
-    counts.write.format("csv").save(output)
 
-    val wc = spark.read.format("csv").load(output)
+    for (fmt <- Seq("csv", "parquet", "json", "orc")) {
+      val fname = "%s.%s".format(output, fmt)
+      counts.write.format(fmt).save(fname)
+    }
+
+    val wc = spark.read.format("csv").load(output + ".csv")
```

Comment:
This should also be in the loop: we should be able to _read_ (not just _write) multiple formats too.